src/SearchInput/components/NoOptions.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { FC, ReactNode, useEffect } from 'react'
+
+interface Props {
+  onNotFound?: () => void
+  notFoundComponent?: ReactNode
+}
+
+export const EmptyResults: FC<Props> = ({ onNotFound, notFoundComponent }) => {
+  useEffect(() => {
+    onNotFound && onNotFound()
```

Comment:
I'm not sure if the new ESLint rule would error on this but you can use optional chaining here

```suggestion
    onNotFound.?()
```